### PR TITLE
Remove tools cache in github runner

### PR DESCRIPTION
**Pull request by praveenrewar**

### CODE REVIEW

File: .github/workflows/test-gh.yml [hunk 1]
```diff
@@ -20,6 +20,12 @@ jobs:
     runs-on: ubuntu-latest
     environment: DockerHub E2E
     steps:
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@main
```
Comment (joaopapereira) [inline .github/workflows/test-gh.yml]: Can we change this to be a specific sha? just to make sure we are always using the same version.

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Let us change the current action from the branch to a specific SHA to ensure we always use the same version.

File: .github/workflows/test-gh.yml (hunk 1 above)
Comment (praveenrewar) [reply .github/workflows/test-gh.yml]: Switched to using the latest released version instead of main.

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**